src/SimplePDPService.sol

```diff
@@ -211,12 +211,16 @@ contract SimplePDPService is PDPListener, PDPRecordKeeper, Initializable, UUPSUp
             revert("Next challenge epoch must fall within the next challenge window");
         }
         uint256 faultPeriods = periodsSkipped;
-        if (!provenThisPeriod[proofSetId]) { 
-            // include previous unproven period 
+        if (!provenThisPeriod[proofSetId]) {
+            // include previous unproven period
             faultPeriods += 1;
         }
         if (faultPeriods > 0) {
-            emit FaultRecord(faultPeriods);
+            uint256[] memory faultedPeriodsList = new uint256[](faultPeriods);
```

Comment:
I don't think this is good.  This event should be constant sized.  It is trivial to reconstruct the faultedPeriodsList offchain from constant size.  And then there's no issues with the event growing very big when recovering from long standing faults.  We should do (provingDeadlines[proofSetId], faultPeriods) or (nextDeadline, faultPeriods)